manifest: hostap: Enable server certificate verification

D-Triveni opened the pull request:
Enable hostname validation for server certificate verification.

Fixes #88697

D-Triveni committed:
manifest: hostap: Enable server certificate verification

Enable hostname validation for server certificate verification.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
net: l2: wifi: Handle domain match and suffix match parameters

Add support to handle domain match and suffix match parameters
for proper server certification validation.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

krish2718 commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -723,6 +723,10 @@ struct wifi_connect_req_params {
 	/** Parameter used for frequency band */
 	enum wifi_frequency_bandwidths bandwidth;
+	/** Domain suffix match */
+	const uint8_t *domain_suffix_match;
```
Please add helpful text and also wpa_supplicant supports passing multiple names too, we should mention that too.

krish2718 replied:
also maybe this should be a user friendly name not wpa_supplicant internal name, e.g., 
```suggestion
    /** Domain name in server certificate will be verified if this is provided */
	const uint8_t *server_cert_domain;
```

krish2718 commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -723,6 +723,10 @@ struct wifi_connect_req_params {
 	enum wifi_frequency_bandwidths bandwidth;
+	/** Domain suffix match */
+	const uint8_t *domain_suffix_match;
+	/** Length of domain suffix match, maximum 32 bytes */
```
why max is 32 bytes? FQDN's can be longer esp. multiple names.

krish2718 reviewed (commented):
we can also add `domain_match` to have a stricter subdomain check.

MaochenWang1 commented:
Is domain_match a more suitable parameter for certificate verification than domain_suffix_match?

D-Triveni committed:
doc: wifi: Add server certificate domain validation instructions

Add instructions for verifying the authentication server’s certificate
domain using exact domain match and domain suffix match options.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>
